### ISSUE TITLE
why no ValidationException constructor ?

### ISSUE THREAD

**Issue by Cyrille37** [dwightwatson/validating#161]
Hi,
I would like to know why there is not ValidationException( MessageBag $errors ) constructor ?
We don't usually need it but sometimes it could be great to re-use it out of a _right_ Validation context.

Thanks

**Comment by dwightwatson** [comment]
The constructor is used for the error message string, which is usual for exceptions. The `MessageBag` is then set on a property, as you can see in the [`throwValidationException()` method](https://github.com/dwightwatson/validating/blob/master/src/ValidatingTrait.php#L387-L400). You can then get the `MessageBag` of errors by calling the [`getErrors()` method on the `ValidationException`](https://github.com/dwightwatson/validating/blob/master/src/ValidationException.php#L26-L34). Hope this helps!

**Comment by Cyrille37** [comment]
Thanks.

But like extended Exception permits to add new methods, it permits to add a new constructor ;-)
It's valid OOP fashion, isn't it ?

Like we don't need 3 lines
```
$e = new ValidationException();
$e->setErrors( $errors );
throw $e ;
```
but only one
```
throw new ValidationException( $errors );
```

**Comment by dwightwatson** [comment]
It's valid OOP fashion, sure. The reason I probably didn't go with that option was just to keep consistency with the base `Exception` constructor's signature. I suppose we could instead adjust the `ValidationException` to accept the model and errors as arguments. Though I'm not sure it's worth the effort to save 3 lines of code.

In fact, just having a look through the framework now, I see Laravel has a validation exception contract... `Illuminate\Contracts\Validation\ValidationException`, which I suppose would be nice if we moved in line with. You only mention the errors, but I think it's important to also include the model context. Could do something like this, though it would break backwards compatibility.

```
<?php

namespace Watson\Validating;

use Illuminate\Contracts\Support\MessageProvider;
use Illuminate\Contracts\Validation\ValidationException as BaseValidationException;

class ValidationException extends BaseValidationException
{
    /**
     * The model with validation errors.
     *
     * @var \Illuminate\Database\Eloquent\Model
     */
    protected $model;

    /**
     * Create a new validation exception instance.
     *
     * @param  \Illuminate\Contracts\Support\MessageProvider  $provider
     * @param  \Illuminate\Database\Eloquent\Model            $model
     * @return void
     */
    public function __construct(MessageProvider $provider, Model $model)
    {
        $this->provider = $provider;
        $this->model = $model;
    }

    /**
     * Get the mdoel with validation errorsl.
     *
     * @return \Illuminate\Database\Eloquent\Model
     */
    public function getModel()
    {
        return $this->model;
    }
}
```

**Comment by Cyrille37** [comment]
Yes, that's great.
But as U say, it breaks backward compatibility ... Perhaps 4 a futur release ;-) aligned with a new major Laravel release like 5.3 ?

thanks a lot for this discussion ;-)

**Comment by dwightwatson** [comment]
I've pushed some stuff to `master` that brings the `ValidationException` in line with Laravel's contract. The change would break backwards compatibility (I've removed the setters from the exception - I suppose that setters on an exception are not good practice) but I'm not sure that is a huge concern. I haven't really been following semantic versioning on this repo, so could always pop it out in the next minor release.
